### DISCUSSION TITLE
Case Insensitivity Issue with Tag Suggestions in Wagtail Admin Interface

### DISCUSSION THREAD

**Discussion by molatok2** [wagtail/wagtail#12095] [Q&A]
Hello! I'm studying Wagtail and encountered the following issue when setting up tags.

I added the following models:

```python
@register_snippet
class BlogTag(TagBase):
    free_tagging = False
    name = models.CharField(max_length=25, unique=True)
    slug = models.SlugField(max_length=255, unique=True, blank=True)

    class Meta:
        verbose_name = "blog tag"
        verbose_name_plural = "blog tags"

class TaggedBlog(ItemBase):
    tag = models.ForeignKey(
        BlogTag, related_name="tagged_blogs", on_delete=models.CASCADE
    )
    content_object = ParentalKey(
        'home.BlogPage',
        on_delete=models.CASCADE,
        related_name='tagged_items'
    )
```
I enabled the setting `TAGGIT_CASE_INSENSITIVE = True`.

However, when a user fills in the tags while creating a page, the case is still taken into account - if the user enters a word in lowercase, options from the dropdown list do not appear.

Here is the model, if it matters:

```python
class BlogPage(Page):
    tags = ClusterTaggableManager(through=TaggedBlog, blank=True)
    date = models.DateField("Мы это написали:")
    author = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True, related_name='blog_pages')
    body = StreamField([
        ('heading', blocks.CharBlock(form_classname="title", label="Заголовок")),
        ('paragraph', blocks.RichTextBlock(label="Текст")),
        ('image', ImageChooserBlock(label="Изображение")),
        ('quote', blocks.BlockQuoteBlock(min_length='20', label="Цитата")),
        ('media', EmbedBlock(label="Ссылка на видео")),
    ], verbose_name='Содержимое')

    content_panels = Page.content_panels + [
        FieldPanel('date'),
        FieldPanel('body'),
        FieldPanel('slug', permission='superuser'),
        FieldPanel('tags'),
    ]

    edit_handler = TabbedInterface([
        ObjectList(content_panels, heading='Центр управления постами'),
        ObjectList(Page.promote_panels, heading='Promote', permission='superuser'),
    ])

    def _get_unique_slug(self):
        slug = slugify(self.title)
        unique_slug = slug
        num = 1
        while Page.objects.filter(slug=unique_slug).exists():
            unique_slug = '{}-{}'.format(slug, num)
            num += 1
        return unique_slug

    def save(self, *args, **kwargs):
        self.slug = self._get_unique_slug()
        super().save(*args, **kwargs)

```
Please tell me what I'm doing wrong

**Comment by Ganesh-403** [comment]
@molatok2 From what I can tell, you're not doing anything wrong in your Wagtail configuration.

`TAGGIT_CASE_INSENSITIVE = True` affects **how django-taggit stores and looks up tags** (e.g. preventing duplicate tags that differ only by case), but it does **not** appear to make the Wagtail admin autocomplete perform case-insensitive suggestions. :contentReference[oaicite:0]{index=0}

In fact, your issue sounds identical to an open issue in **django-taggit** discussing case-sensitive autocomplete behavior despite `TAGGIT_CASE_INSENSITIVE` being enabled. :contentReference[oaicite:1]{index=1}

A few things to verify:

1. Confirm that `TAGGIT_CASE_INSENSITIVE = True` is actually being loaded by your Django settings (e.g. via `python manage.py shell`).

2. Check your versions of:
   - Wagtail
   - django-taggit
   - django-modelcluster

3. If you're using `free_tagging = False`, the dropdown suggestions come from the existing `BlogTag` records, but the matching performed by the autocomplete endpoint still appears to be case-sensitive in this scenario. I couldn't find documentation indicating that Wagtail overrides this behavior. :contentReference[oaicite:2]{index=2}

Could you also share:

- your Wagtail version,
- your django-taggit version,
- and whether the problem is only with the **autocomplete suggestions**, or whether saving `"python"` correctly reuses an existing `"Python"` tag instead of creating a duplicate?

That distinction would help determine whether this is purely an autocomplete issue or whether `TAGGIT_CASE_INSENSITIVE` itself isn't taking effect.

Based on what you've described, I'd lean toward this being a limitation/bug in the autocomplete implementation rather than a problem with your models.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
